### PR TITLE
Implement padding for TSMapEstimator

### PR DESCRIPTION
**Pull request by adonath**
**Description**

This pull request implement padding for the `TSMapEstimator` too avoid blank boundaries of the result maps and to be able to work with arbitrary kernel sizes. In addition in restructures a bit the implementation and adds some little helper methods, like:
- `Map.to_unit()`
- `MapAxis.iter_by_edges`

**Dear reviewer**

### CODE REVIEW

File: gammapy/estimators/asmooth_map.py [hunk 1]
```diff
@@ -154,21 +154,23 @@ def run(self, dataset):
...
+            energy_axis = dataset.counts.geom.axes["energy"].squash()
         else:
-            energy_edges = self.energy_edges
+            energy_axis = MapAxis.from_energy_edges(self.energy_edges)
 
         results = []
 
         for energy_min, energy_max in progress_bar(
-            zip(energy_edges[:-1], energy_edges[1:]),
+            energy_axis.iter_by_edges,
```
Comment (bkhelifi) [inline gammapy/estimators/asmooth_map.py:164]: great function, even if harder to read/understand for non-experts

File: gammapy/maps/geom.py [hunk 2]
```diff
@@ -882,6 +882,12 @@ def as_xerr(self):
             self.edges[1:] - self.center,
         )
 
+    @property
+    def iter_by_edges(self):
```
Comment (registerrier) [inline gammapy/maps/geom.py:886]: OK Why not.

File: gammapy/estimators/tests/test_ts_map.py [hunk 3]
```diff
@@ -251,5 +242,5 @@ def test_ts_map_with_model(fake_dataset):
         energy_edges=[200, 3500] * u.GeV,
     )
     maps = estimator.run(fake_dataset)
-    assert_allclose(maps["sqrt_ts"].data[:, 25, 25], 0.323, atol=0.1)
-    assert_allclose(maps["flux"].data[:, 25, 25], 1.015417e-12, atol=1e-12)
+    assert_allclose(maps["sqrt_ts"].data[:, 25, 25], -0.378473, atol=0.1)
```
Comment (registerrier) [inline gammapy/estimators/tests/test_ts_map.py:245]: Why the change here?

**Review by registerrier — APPROVED** [review]
Thanks @adonath . This looks good.

I left some minor  inline comments.

File: gammapy/estimators/ts_map.py [hunk 4]
```diff
@@ -184,6 +182,15 @@ def selection_all(self):
 
         return selection
 
+    def energy_edges_axis(self, dataset):
```
[the remark this answers is not in the document]
Comment (adonath) [reply gammapy/estimators/ts_map.py:185]: I agree, there is no strong need to have it public...

[the remark this answers is not in the document]
Comment (adonath) [reply gammapy/estimators/ts_map.py]: Done

File: gammapy/estimators/tests/test_ts_map.py (hunk 3 above)
Comment (adonath) [reply gammapy/estimators/tests/test_ts_map.py]: Fixed

File: gammapy/maps/geom.py (hunk 2 above)
Comment (adonath) [reply gammapy/maps/geom.py:886]: I thought it's a very common pattern in Gammapy I can adapt the ~5 other cases in a follow up PR....

**Comment by adonath** [comment]
Thanks @registerrier!
